### PR TITLE
Fix integer overflow in replication_pad2d for large padding values

### LINKED ISSUE #169741: ReplicationPad2dwith extremely large padding value causes segmentation fault
### 🐛 Describe the bug

nn.ReplicationPad2dtriggers a segmentation fault when given extremely large padding values (near INT64_MAX), rather than raising a proper exception.

### Expected Behavior:
Should raise a ValueErroror RuntimeErrorindicating the padding value is too large or invalid.
### Actual Behavior:
Segmentation fault (program crashes).
### Root Cause:
The extremely large padding value likely causes integer overflow in internal size calculations, leading to invalid memory accesses in C++ backend.

```python
import torch
from torch.nn import ReplicationPad2d
input_data = torch.randn(1, 3, 10, 10)
pad = ReplicationPad2d(9223372036854775807)
pad.register_module(name='replication_pad', module=pad)
output_data = pad(input_data)
```
### Versions

PyTorch version: 2.7.1+cu128
Is debug build: False
CUDA used to build PyTorch: 12.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.0 (default, Mar 3 2022, 09:58:08) [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.15.0-122-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 11.5.119
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration:
GPU 0: NVIDIA RTX A6000

Nvidia driver version: 535.183.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture: x86_64
CPU op-mode(s): 32-bit, 64-bit
Address sizes: 43 bits physical, 48 bits virtual
Byte Order: Little Endian
CPU(s): 64
On-line CPU(s) list: 0-63
Vendor ID: AuthenticAMD
Model name: AMD EPYC 7282 16-Core Processor
CPU family: 23
Model: 49
Thread(s) per core: 2
Core(s) per socket: 16
Socket(s): 2
Stepping: 0
Frequency boost: enabled
CPU max MHz: 2800.0000
CPU min MHz: 1500.0000
BogoMIPS: 5600.23
Flags: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip rdpid overflow_recov succor smca sme sev sev_es
Virtualization: AMD-V
L1d cache: 1 MiB (32 instances)
L1i cache: 1 MiB (32 instances)
L2 cache: 16 MiB (32 instances)
L3 cache: 128 MiB (8 instances)
NUMA node(s): 2
NUMA node0 CPU(s): 0-15,32-47
NUMA node1 CPU(s): 16-31,48-63
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit: Not affected
Vulnerability L1tf: Not affected
Vulnerability Mds: Not affected
Vulnerability Meltdown: Not affected
Vulnerability Mmio stale data: Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed: Mitigation; untrained return thunk; SMT enabled with STIBP protection
Vulnerability Spec rstack overflow: Mitigation; safe RET
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1: Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2: Mitigation; Retpolines; IBPB conditional; STIBP always-on; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds: Not affected
Vulnerability Tsx async abort: Not affected

Versions of relevant libraries:
[pip3] numpy==2.2.6
[pip3] torch==2.7.1+cu128
[pip3] torchaudio==2.7.1+cu128
[pip3] torchvision==0.22.1+cu128
[pip3] triton==3.3.1
[conda] numpy 2.2.6 pypi_0 pypi
[conda] torch 2.7.1+cu128 pypi_0 pypi
[conda] torchaudio 2.7.1+cu128 pypi_0 pypi
[conda] torchvision 0.22.1+cu128 pypi_0 pypi
[conda] triton 3.3.1 pypi_0 pypi

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

### PR DESCRIPTION
**Pull request by Neuro1729**
Add explicit checks to prevent integer overflow when computing output height  and width in replication_pad2d. Previously, extremely large padding values  could cause oheight or owidth to exceed INT64_MAX, resulting in undefined  behavior or crashes.

Changes:
- Added TORCH_CHECK to verify that pad_t + pad_b does not overflow iheight.
- Added TORCH_CHECK to verify that pad_l + pad_r does not overflow iwidth.

This ensures that the function fails safely with a descriptive error  message instead of crashing when unrealistic padding values are provided.

Fixes #169741

### CODE REVIEW

File: aten/src/ATen/native/ReplicationPadding.cpp [hunk 1]
```diff
@@ -108,6 +109,28 @@ TORCH_META_FUNC(replication_pad2d) (
   int64_t nslices = input.size(dimslices);
   int64_t iheight = input.size(dimh);
   int64_t iwidth = input.size(dimw);
+  
+  const int64_t INT64_MAX = std::numeric_limits<int64_t>::max();
+  
+  int64_t safe_pad_t = std::max(pad_t, (int64_t)0);
+  int64_t safe_pad_b = std::max(pad_b, (int64_t)0);
+  int64_t safe_pad_l = std::max(pad_l, (int64_t)0);
+  int64_t safe_pad_r = std::max(pad_r, (int64_t)0);
```
Comment (cyyever) [inline aten/src/ATen/native/ReplicationPadding.cpp]: What if they are < 0?

File: aten/src/ATen/native/ReplicationPadding.cpp [hunk 2]
```diff
@@ -108,6 +109,28 @@ TORCH_META_FUNC(replication_pad2d) (
...
+  
+  TORCH_CHECK(iheight <= INT64_MAX - safe_pad_t,
+    "Padding value ", pad_t, " for height is too large and would cause int64 overflow.");
+  int64_t temp_height = iheight + safe_pad_t;
+
+  TORCH_CHECK(temp_height <= INT64_MAX - safe_pad_b,
+    "Padding value ", pad_b, " for height is too large and would cause int64 overflow.");
```
Comment (albanD) [inline aten/src/ATen/native/ReplicationPadding.cpp:125]: Why not do this in a single check for iheight <= INT64_MAX - safe_pad_t - safe_pad_b ?

Same below

Comment (Neuro1729) [reply aten/src/ATen/native/ReplicationPadding.cpp:125]: I wrote it this way to be explicitly defensive against intermediate overflows. If we do INT64_MAX - safe_pad_t - safe_pad_b, we are safe as long as the pads are positive, but the step-by-step approach ensures we never perform a calculation that exceeds INT64_MAX at any point in the logic. Do you prefer the more concise version for consistency with other padding functions?
